oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/query/DurationUtils.java

```diff
@@ -216,4 +216,59 @@ private DateTime parseToDateTime(Downsampling downsampling, long time) throws Pa
 
         return dateTime;
     }
+
+    public long convertBucketTotIimestamp(boolean isStart, long bucket) throws ParseException {
+        String bucketStr = String.valueOf(bucket);
+        if (bucketStr.length() < 14) {
```

Comment:
What is this for?

Reply:
According to the existing conversion rules, there is a short time such as 20200115

Reply:
Still not following. Timebucket always could be determined by length, why separate by 14?